ScanShop.Mobile/ScanShop.Mobile/App.xaml.cs

```diff
@@ -14,17 +13,9 @@ public App()
 
             MainPage = new AppShell();
 
-            if (string.IsNullOrEmpty(SecureStorage.GetAsync("BearerToken").Result))
-            {
-                Shell.Current.GoToAsync("//LoginPage");
-            }
-            else
-            {
-                var bearerToken = SecureStorage.GetAsync("BearerToken");
-                var httpClientService = DependencyService.Get<IHttpClientService>();
-                httpClientService.SetBearerToken(bearerToken.Result);
-                Shell.Current.GoToAsync("//OrdersPage");
-            }
+            var httpClientService = DependencyService.Get<IHttpClientService>();
+            httpClientService.InitializeAsync();
```

Comment:
Never leave Async method unfinished if it is not supposed to. Use synchronous .Wait() or .Result

Reply:
Fixed in new commit